theano/compile/sharedvalue.py

```diff
@@ -120,6 +120,36 @@ def set_value(self, new_value, borrow=False):
         Changes to this value will be visible to all functions using
         this SharedVariable.
 
+        Notes
+        -----
+        Prior to Theano 0.3.1, set_value did not work in-place on the GPU. This
+        meant that sometimes, GPU memory for the new value would be allocated
+        before the old memory was released. If you're running near the limits of
+        GPU memory, this could cause you to run out of GPU memory.
+
+        Beginning with Theano 0.3.1, set_value will work in-place on the GPU, if
```

Comment:
The new backend never moves data in-place when doing set_value on a shared variable.  But this could be added.